### PR TITLE
gateway-azure: verify md5sum sent by the client

### LINKED ISSUE #4289: [gateway-azure]: PutObject is not verifying md5sum
```
func (a AzureObjects) PutObject(bucket, object string, size int64, data io.Reader, metadata map[string]string, sha256sum string) (objInfo ObjectInfo, err error) {                       
        var sha256Writer hash.Hash                                                                                                                                                       
        teeReader := data                                                                                                                                                                
        if sha256sum != "" {                                                                                                                                                             
                sha256Writer = sha256.New()                                                                                                                                              
                teeReader = io.TeeReader(data, sha256Writer)                                                                                                                             
        }                                                                                                                                                                                
                                                                                                                                                                                         
        delete(metadata, "md5Sum")
```

Azure gateway is ignoring the md5sum sent by client. It should verify that it is correct.

### PR DESCRIPTION
**Pull request by krishnasrinivas**
fixes #4289

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by harshavardhana — DISMISSED** [review]

**Comment by harshavardhana** [comment]
Builds are failing for this @krishnasrinivas can you fix them ?

**Comment by hackintoshrao** [comment]
@krishnasrinivas : Can you Md5 verification test to `server_test.go`, will be soon updating Mint to keep them in sync.

**Comment by krishnasrinivas** [comment]
@hackintoshrao server_test.go does not test gateway, currently mint is the only way to test gateway.

@harshavardhana @vadmeste can you guys review?

File: cmd/gateway-azure.go [hunk 1]
```diff
@@ -313,24 +314,47 @@ func canonicalMetadata(metadata map[string]string) (canonical map[string]string)
 // uses Azure equivalent CreateBlockBlobFromReader.
 func (a *azureObjects) PutObject(bucket, object string, size int64, data io.Reader, metadata map[string]string, sha256sum string) (objInfo ObjectInfo, err error) {
 	var sha256Writer hash.Hash
+	var md5sumWriter hash.Hash
+
+	var writers []io.Writer
+
+	md5sum := metadata["md5Sum"]
```
Comment (vadmeste) [inline cmd/gateway-azure.go]: @krishnasrinivas, recently md5Sum was changed to etag. You need to update this.

Comment (krishnasrinivas) [reply cmd/gateway-azure.go]: @vadmeste fixed and pushed commit

**Review by vadmeste — APPROVED** [review]
Tested with corrupted md5

**Comment by krishnasrinivas** [comment]
https://github.com/minio/minio/pull/4385 superceeds this patch

**Pull request closed without merging**